### PR TITLE
[MRG+1] Remove warnings in nested Parallel when the inner Parallel has n_jobs=1

### PR DESCRIPTION
**Pull request by harishmk**
suppress a warning if a daemonic process creates parallel jobs with n_jobs=1
#405

### CODE REVIEW

**Comment by GaelVaroquaux** [comment]
LGTM. +1 for merge. Thank you!

**Comment by aabadie** [comment]
I could reproduce the initial problem mentionned in #405 on Windows with python 2.7. The fix looks good to me.
One thing is missing though : a test. Can you add a test function in joblib/test/test_parallel.py that checks no warning is raised when n_jobs = 1 ?

**Comment by lesteve** [comment]
A snippet reproducing the problem:

```py
from joblib import Parallel, delayed


def func():
    return 42


def inner_parallel_func():
    Parallel(n_jobs=1)(delayed(func)() for _ in range(10))

if __name__ == '__main__':
    Parallel(n_jobs=2)(delayed(inner_parallel_func)() for _ in range(10))
```

**Comment by lesteve** [comment]
I used this as an excuse to test the recent features of github where PR authors can allow the maintainers to push commits on the PR branch and it seems to work just fine, this is very nifty!

I added a test for the warnings, as you can see from the diff it's not that easy to catch the warnings because they happen in child processes so I reuse and enhanced check_subprocess_call ...

**Comment by lesteve** [comment]
@aabadie do you want to have a look at this one before I merged?

File: joblib/test/test_parallel.py [hunk 1]
```diff
@@ -768,3 +769,42 @@ def generate_arrays(n):
...
+from joblib import Parallel, delayed
+
+
+def func():
+    return 42
+
+
+def parallel_func():
+    res =  Parallel(n_jobs={inner_n_jobs})(delayed(func)() for _ in range(3))
+    # Needed otherwise some warnings may be missed
```
Comment (aabadie) [inline joblib/test/test_parallel.py]: not super critical but I think this comment is useless or there's something missing after.

**Comment by lesteve** [comment]
> I admit adding a test for this was a bit tricky :)

I was quite happy that check_subprocess_call (originally here for checking that a function defined in `__main__` can be used inside `Parallel`, which is very useful in interactive i.e. IPython settings) could be reused without too much pain for the test.

**Comment by aabadie** [comment]
Merging this one. Thanks @harishmk and @lesteve

**Comment by lesteve** [comment]
@harishmk do you mind giving me your full name ?

I would need it for CHANGES.rst since I am planning to do a bug-fix release of joblib (mainly for scikit-learn 0.18.1) and this PR would be part of it. If I don't hear from you, I'll just use your github user name.

**Pull request merged**